server/plugins/TriggerUGens.cpp

```diff
@@ -895,23 +897,67 @@ void SetResetFF_next_k(SetResetFF* unit, int inNumSamples) {
 
 
 void ToggleFF_Ctor(ToggleFF* unit) {
-    SETCALC(ToggleFF_next);
+    if (unit->mCalcRate == calc_FullRate) {
+        if (INRATE(0) == calc_FullRate) {
+            SETCALC(ToggleFF_next);
+        } else {
+            SETCALC(ToggleFF_next_ak);
+        }
+    } else {
+        if (INRATE(0) == calc_FullRate) {
+            SETCALC(ToggleFF_next_ka);
+        } else {
+            SETCALC(ToggleFF_next);
+        }
+    }
 
     unit->m_prevtrig = 0.f;
     unit->mLevel = 0.f;
 
-    ZOUT0(0) = 0.f;
+    ToggleFF_next(unit, 1);
+    // no need to reset internal state
```

Comment:
I see why it's not necessary to reset the state because of how the toggling behaves. But, I think it would still be best to stick with the pattern of resetting the state after the init sample is calculated. Strictly speaking, the _state_ is inaccurate at the time the first sample is calculated in this case, i.e. `m_prevtrig` remains `1.0`, when in principle it should be `0.0` (same with level, which "should" be `0.0`) at the start of the first calculation block.

It's really a matter of consistency and to reinforce the proper pattern. Open to arguments otherwise though!

Reply:
I'm a bit conflicted about this:
- repeating calculation is not a big deal, I think it would be worth it to reinforce the pattern. However it's also not necessary and not repeating it is one less change to how the UGen behaves (at least in theory, probably negligible in practice)
- we recently had another case of skipping reset in #6187 with IEnvGen, where in the end we decided to put a comment... and in TriggerUGen there are many more cases, where y(0) is computed at Ctor and "cached" for the first _next call. 

It would be nice to decide which way we wanna go. Maybe IEnvGen stays as an exception, and all TriggerUGens should reset properly even if they can easily "cache" y(0)?

Reply:
Given how pervasive this initialization sample problem is, my vote is to follow the pattern. 

I have a sense that the problem took root because UGens were added that basically copied the Ctor of others, which weren't explicit about handling initial state.

I also think that having state cached between init and the first calc block is harder to reason about and obscures the important distinction between and init sample and what is output at runtime.

Reply:
Ok, since I'm working on more plugins from TriggerUGens, I think it will be better to look at them all together. I'll remove the change to initialization from this PR and open another one with all of them